storage/client_raft_test.go

```diff
@@ -239,7 +241,9 @@ func TestFailedReplicaChange(t *testing.T) {
 	}()
 	storage.TestingCommandFilter = func(method string, args proto.Request, reply proto.Response) bool {
 		if method == proto.EndTransaction && args.(*proto.EndTransactionRequest).Commit == true {
-			reply.Header().SetGoError(util.Errorf("boom"))
+			oldErr := unsafe.Pointer(&reply.Header().Error)
+			newErr := proto.Error{Message: "boom"}
+			atomic.SwapPointer((*unsafe.Pointer)(oldErr), unsafe.Pointer(&newErr))
```

Comment:
Google only led me to http://play.golang.org/p/7Q5B7jE-uz via https://groups.google.com/forum/#!topic/golang-nuts/2Abz-DnSy8w and they do the same thing. Would love to know why that's how it works and not differently.